lib/huginn_acumen_product_agent/acumen_product_agent.rb

```diff
@@ -256,14 +256,19 @@ def map_attributes(product)
         end
 
         def update_availability(product)
-          stock_quantity = product['acumenAttributes']['stock_quantity']
-          publication_date = product['datePublished']
-          stock_quantity = stock_quantity.present? ? stock_quantity.to_i : 0
-
-
-          if (!product['isDigital'] && product['productAvailability'] == 'available' && stock_quantity < 1)
-            product['productAvailability'] = publication_date && publication_date.to_datetime > DateTime.current().end_of_day ? 'preorder' : 'disabled'
-          end
+            stock_quantity = product['acumenAttributes']['stock_quantity']
+            publication_date = product['datePublished']
+            no_backorder_fill = product['noBackorderFill']
+            stock_quantity = stock_quantity.present? ? stock_quantity.to_i : 0
+
+            if (!product['isDigital'] && product['productAvailability'] == 'available')
+                if ((publication_date && publication_date.to_datetime > DateTime.current().end_of_day) || (!no_backorder_fill && stock_quantity < 1))
+                    product['productAvailability'] = 'preorder'
+                end
+                if (no_backorder_fill && stock_quantity < 1)
```

Comment:
Consider adding a blank line here. It's not strictly necessary, but it is easier to read.